calculator/main.py

```diff
@@ -26,7 +32,26 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    if not params or not isinstance(params, tuple):
+        raise InvalidParams("Given params are invalid.")
+    for param in params:
+        if not isinstance(param, (int, float)): 
+            raise InvalidParams("Given params are invalid.")
+    
+    if operation not in get_operations(calc):
+          raise InvalidOperation("Given operation is invalid.")
+
+    
+    result =  calc['operations'][operation](*params)
+    t_stamp = datetime.now()
+    t_stamp = t_stamp.strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
Instead of setting `t_stamp` to `datetime.now()` and then replacing it on the next line you could just do `t_stamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')`